security/pkg/pki/ca/ca.go

```diff
@@ -78,6 +79,25 @@ type IstioCA struct {
 	livenessProbe  *probe.Probe
 }
 
+// LivenessCheckController updates the availability of the liveness probe of the CA instance
+type LivenessCheckController struct {
+	interval time.Duration
+	ca       *IstioCA
+}
+
+// Run starts the check routine
+func (c *LivenessCheckController) Run() {
+	go func() {
+		t := time.NewTicker(c.interval)
+		for {
+			select {
+			case <-t.C:
+				c.ca.livenessProbe.SetAvailable(c.ca.verify())
```

Comment:
So you are relying on the ca.verify() to verify the signing cert, maybe it's better to have a dedicated function or class to do this verification? I know we will rely on the gRPC server loopback call to determine the liveness as well.

Reply:
Used grpc.client.SendCSR to check server liveness